app/models/metric/capture.rb

```diff
@@ -38,60 +38,55 @@ def self.alert_capture_threshold(target)
                 ::Settings.performance.capture_threshold_with_alerts.default)
   end
 
+  # legacy messages on the queue
+  # went away in 4.6
   def self.perf_capture_timer(zone = nil)
-    _log.info("Queueing performance capture...")
-
-    zone ||= MiqServer.my_server.zone
-    perf_capture_health_check(zone)
-    targets = Metric::Targets.capture_targets(zone)
-
-    targets_by_rollup_parent = calc_targets_by_rollup_parent(targets)
-    target_options = calc_target_options(zone, targets_by_rollup_parent)
-    targets = filter_perf_capture_now(targets, target_options)
-    queue_captures(targets, target_options)
+    zone ||= MiqServer.my_server(true).zone
+    zone.ext_management_systems.each do |ems|
+      perf_collect_all_metrics_queue(ems, "realtime")
+    end
+  end
 
-    # Purge tasks older than 4 hours
-    MiqTask.delete_older(4.hours.ago.utc, "name LIKE 'Performance rollup for %'")
+  def self.perf_collect_all_metrics_queue(ems, interval_name = "realtime", start_time = nil, end_time = nil, options = nil)
+    Array.wrap(ems).each do |e|
+      MiqQueue.put_unless_exists(
+        :class_name  => "Metric::Capture",
+        :method_name => "perf_collect_all_metrics",
+        :args        => [e.id, interval_name, start_time, end_time, options],
+        :role        => "ems_metrics_collector",
+        :queue_name  => e.metrics_collector_queue_name,
+        :zone        => e.zone_id,
+        :priority    => MiqQueue::HIGH_PRIORITY,
+        :state       => %w(ready dequeue),
+      )
+    end
+  end
 
-    _log.info("Queueing performance capture...Complete")
+  def self.perf_collect_all_metrics(ems, interval_name = "realtime", start_time = nil, end_time = nil, options = nil)
+    ems = ExtManagementSystem.find(ems) unless ems.kind_of?(ExtManagementSystem)
+    klass = ems.class::MetricsCapture rescue ManageIQ::Providers::BaseManager::MetricsCapture
```

Comment:
I don't think we need to worry about this, if an EMS doesn't have a `MetricsCapture` we will just blowup later in `just_perf_capture` when calling `perf_collect_metrics`

Reply:
was thinking this gave us compatibility for all providers while we rolled out new code

Since this is going into older branch, thought we'd go for minimal change to each of those repos

Reply:
fixed